jsapp/js/projects/projectsTable/colResize.tsx

```diff
@@ -0,0 +1,271 @@
+import React, {useState, useEffect, useReducer, useRef} from 'react';
+
+/**
+ * Creates and updates styles for column widths based on drag events on resize handles.
+ * Most of the logic is in the the event handler function.
+ *
+ * Usage notes
+ * - The SCSS for the table cell widths define the default widths.
+ * - When dragging a resize handle, this component overrides a given field
+ *   with a fixed width in px units.
+ * - Header and row cells use [data-field={field.name}]
+ * - Resizers, with [data-resize-fieldname={field.name}], are a direct child of
+ *   a table header cell.
+ *
+ * Ideas
+ * - Tweaking the default minimums/maximums
+ *   - There's a console.log you can uncomment under mouseup that can help
+ *     with choosing widths.
+ *   - I changed a few just for example, based on what looked okay to me in
+ *     English.
+ *   - Dates truncate awkwardly, so I gave them more width.
+ *   - Not in scope, but it might be possible to compute good defaults during
+ *     initialization, from text widths of localized headings.
+ * - Consequence of fluid widths
+ *   - Try resizing different columns in different orders, sometimes neighbors
+ *     adapt in unexpected ways.
+ *   - Could use hard px values for defaults to avoid.
+ * - Persistence
+ *   - Widths are cleared on refresh (or unmount)
+ *   - Future, could save them in localstorage, clearing on logout.
+ * - Touch events
+ *   - Haven't tested kpi on a touchscreen recently, but the code is adaptable.
+ *   - Might be pretty simple to support by switching to pointer events.
+ */
+
+/**
+ * A flexible subset of fieldnames mapped to widths in px.
+ * Example:
+ * {
+ *   name:   100,  //  the "name" column will be 100px wide
+ *   status: 150  // the "status" column will be 150px wide
+ * }
+ */
+interface ColumnWidths {
+  [key: string]: number;
+}
+interface ColumnWidthReducerAction {
+  type: 'resize';
+  column: string;
+  width: number;
+}
+/**
+ * Reducer for column width updates.
+ * Almost too simple to justify its existence.
+ *
+ * Pulls a tiny bit of update logic out of the event handlers.
+ */
+function columnWidthReducer(
+  columnWidths: ColumnWidths,
+  action: ColumnWidthReducerAction
+) {
+  switch (action.type) {
+    case 'resize':
+      return {
+        ...columnWidths,
+        [action.column]: action.width,
+      };
+    default:
+      return {...columnWidths};
+  }
+}
+
+/**
+ * The minimum resizing width for a given column.
+ *
+ * This prevents columns from being entirely cropped away.
+ * Currently a minimum of 108px.
+ * Can be hardcoded per-field.
+ */
+function minColWidth(fieldname: string) {
+  const defaultMinimumWidth = 108;
+  const minimums: ColumnWidths = {
+    countries: 116,
+    dateModified: 144,
+    dateDeployed: 144,
+  };
```

Comment:
Decided:

- Keep the min-widths in the ColumnResizing file for now (instead of mixing them in with other field constants)
   - Easier to see, e.g., the only 3 field exceptions that are defined so far
   - Make it easier to define more following the pattern without jumping/scrolling in multiple files
- [ ] Use type ProjectFieldName instead of generic 'string'

(via internal discussion)